lib/hammer_cli/defaults.rb

```diff
@@ -0,0 +1,120 @@
+require 'hammer_cli'
+require 'yaml'
+module HammerCLI
+
+  class DefaultsCommand < HammerCLI::AbstractCommand
+
+    class ListDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'list'
+
+      desc 'List all the defaults parameters'
+      def execute
+        list_all_defaults_message
+        HammerCLI::EX_OK
+      end
+    end
+
+    class AddDefaultsCommand < HammerCLI::DefaultsCommand
+      command_name 'add'
+
+      desc 'Add a default parameter to config'
+      option "--param-name", "option-name", _("The name of the default option"), :required => true
+      option "--param-val", "option-value", _("The value for the default option")
+      option "--plugin-name", "option-name", _("The plugin name defaults will be generated for")
```

Comment:
For all options above `"option-name"` should be capital. Also please use underscores instead of dashes.